test/functional/combine_logs.py

```diff
@@ -76,9 +77,17 @@ def read_logs(tmp_dir):
     Delegates to generator function get_log_events() to provide individual log events
     for each of the input log files."""
 
+    # Find out what the folder is called that holds the debug.log file
+    chain = glob.glob("{}/node0/*/debug.log".format(tmp_dir))
+    if chain:
+        chain = chain[0]  # pick the first one if more than one chain was found (should never happen)
+        chain = chain.split('/node0/')[-1].split('/debug.log')[0]  # extract the chain name
+    else:
+        chain = 'regtest'  # fallback to regtest (should only happen when none exists)
```

Comment:
Maybe
```Python
chain = re.search('node0/(.+?)/debug', chain[0]).group(1) if chain else 'regtest'
```

Reply:
I like the multi-line version because it allows me to add comments. Though, I am happy to use the `re` solution if others think it is more readable/worthwhile.

Reply:
You could also just do L84 as
```Python
chain = re.search('node0/(.+?)/debug', chain).group(1)
```